content/docs/examples/creds-in-ocmconfig.md

```diff
@@ -0,0 +1,225 @@
+---
+title: "Credentials in ocmconfig file"
+description: ""
+summary: ""
+date: 2024-09-04T13:54:01+02:00
+lastmod: 2024-09-04T13:54:01+02:00
+draft: false
+weight: 999
+toc: true
+---
+
+## Overview
+
+The [OCM command line client](https://github.com/open-component-model/ocm/blob/main/docs/reference/ocm.md) can be configured by supplying it with a [configuration file](https://github.com/open-component-model/ocm/blob/main/docs/reference/ocm_configfile.md). By default, the CLI looks for configuration in `$HOME/.ocmconfig`, if it exists.
+
+The configuration file can be used in particular to specify the credentials, which are required for the CLI to be able to access the artifact repositories, referenced in CLI commands.
+
+## Examples
+
+This page contains basic examples of credentials configuration for a few most common artifact repository types. The examples below are complete `.ocmconfig` files, not snippets.
+
+For comprehensive documentation on the credentials topic, including usage of certificates or HashiCorp Vault, execute the command `ocm credential-handling`.
+
+### Repositories and Consumers
+
+In the examples below some configuration is located under `configurations[0]`.`repositories`, and some other under `configurations[0]`.`consumers`. This chanpter explains the difference between `repositories` and `consumers`, which is potentially not as obvious as one could think.
+
+In this context, `repository` is a place, where credentials can be stored, i.e. it is a credentials repository. For example, Docker's `config.json` can store multiple credentials, and in that sense the file serves as a repository that can store and provide credentials. That is why its location is configured under `repositories`. Other examples of credentials repositories can be the NPM's `.npmrc` file or a HashiCorp Vault instance.
+
+A `consumer` is something the credentials are required for. For example, if you need to configure credentials that are required to log in to an OCI registry, one could say that the registry will be consuming these credentials, i.e. the registry is a credentials consumer. That is why it is configured under `consumers`.
+
+### Re-use credentials configured for Docker
+
+This `.ocmconfig` file will tell OCM CLI to use credentials configuration from Docker's `config.json` file.
+
+```yaml
+type: generic.config.ocm.software/v1
+configurations:
+  - type: credentials.config.ocm.software
+    repositories:
+      - repository:
+          type: DockerConfig/v1
+          dockerConfigFile: "~/.docker/config.json"
+          propagateConsumerIdentity: true

```

Comment:
After having a quick glance at our library code, it appears as if our `propagateConsumerIdentity` behaviour is quite inconsistent. 
Every credential repository spec may contain this property. But while e.g. _dockerconfig_ defaults `propagateConsumerIdentity` to `false` if it is not specified, _npm_  defaults `propagateConsumerIdentity` to `true`, if it is not specified. As to my understanding, this property has to be set to true in order to be actually useful for _command line usage_. That's why it is required here!
We should align this for all credential repositories and I'd argue we should default to true, since command line usage is the primary use case for the config file. Furthermore, I think a common ocm cli user should not have to bother trying to understand this property.

Reply:
I was wrong, it is defaulted for dockerconfig as well!

Reply:
What is your suggestion? Uwe's documentation has several examples with `propagateConsumerIdentity: true`, e.g. [here](https://github.com/open-component-model/ocm/blob/main/docs/reference/ocm_configfile.md). I'd prefer to keep the line, to have it consistent. Second, the purpose of this page is to provide a working example, and in this form the file was working for my use cases.

Reply:
Suggestion to remove `propagateConsumerIdentity: true` accepted